src/SFA.DAS.Employer.PR.Web/Controllers/SetPermissionsController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Domain.Interfaces;
+using SFA.DAS.Employer.PR.Domain.Models;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Permissions;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using SFA.DAS.Employer.PR.Web.Constants;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+using SFA.DAS.Employer.PR.Web.Models;
+using SFA.DAS.Employer.PR.Web.Models.Session;
+using System.Security.Claims;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
+[Route("accounts/{employerAccountId}/providers/new/setPermissions", Name = RouteNames.SetPermissions)]
+public class SetPermissionsController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<SetPermissionsSubmitViewModel> _validator) : Controller
+{
+    [HttpGet]
+    public IActionResult Index([FromRoute] string employerAccountId)
+    {
+        var model = GetViewModel(employerAccountId);
```

Comment:
Session model check is missing?

Reply:
no the method GetViewModel has the check and returns null if there's an issue, which then redirects

Reply:
ok